Restructuring and adding icons

sneddobuilds opened the pull request:
Restructured into the standard grapesjs plugin template(ish). 

Also added custom icons for column and row top level menu items.

sneddobuilds committed:
Restructuring and adding icons

VojtechLanka commented:
Thanks. I'm currently busy with another project, but I will try to check this by the end of the week.

sneddobuilds commented:
No rush, I am using based off my fork atm, but as all the clever stuff is yours I wanted to give you a pull req :)

sneddobuilds committed:
Create npm-publish.yml

sneddobuilds committed:
Update npm-publish.yml

sneddobuilds committed:
Update npm-publish.yml

sneddobuilds committed:
Initial changes to allow headers. Added a toolbar command for the setting modal.

sneddobuilds committed:
Finishing headers and fixing some styling.

#1 #2

sneddobuilds committed:
Merge pull request #4 from sneddobuilds/TableHeaders

Table headers

VojtechLanka commented on src/commands/index.js:
```diff
@@ -0,0 +1,426 @@
+      let settingExists = tb.find(o=> o.command === 'open-table-settings-modal');
+      if(!settingExists)
+      {
+        tb.push(
```
The command `open-table-settings-modal` is not being added for me.

sneddobuilds replied:
What browser are you using? Tested this on my side and it seems to be working. I'm testing on windows in chrome/edge.

sneddobuilds replied:
It should get added to the table element only. This can be a pain to select sometimes, I usually get it via the arrows.

VojtechLanka replied:
Found the issue it was on my end. There was pice of code I have forgotten about that either set custom toolbar if it was custom element or set the default one so it was just overwritten now the command is added correctly.

VojtechLanka commented on src/commands/index.js:
```diff
@@ -0,0 +1,426 @@
+      const tb = component.get('toolbar');
+      let settingExists = tb.find(o=> o.command === 'open-table-settings-modal');
+      if(!settingExists)
+      {
```
I do not like that some ifs have the bracket on newline while some don't. I think only one style should be used.

sneddobuilds replied:
Updated eslint to highlight these issues and fixed them

VojtechLanka commented on src/commands/index.js:
```diff
@@ -0,0 +1,426 @@
+              attributes: {class: 'fa fa-cog', title: 'Settings'},
+          }
+        );
+        return tb;
```
Seems like unneeded return.

sneddobuilds replied:
indeed.

VojtechLanka commented on src/components/customTable.js:
```diff
@@ -0,0 +1,105 @@
+        attributes: {
+          'data-n_columns': 5,
+          'data-n_rows': 3,
+          'data-hasHeader': false,
```
I don't like that attributes don't have the same case.

VojtechLanka commented on src/components/customTable.js:
```diff
@@ -0,0 +1,105 @@
+      //     attributes: {class: 'fa fa-cog', title: 'Settings'},
+      //   });
+      //   this.set('toolbar', tb);
+      // },
```
I don't think there is reason to keep this disabled code here, if it will be needed it should be possible to find in git commit.

sneddobuilds replied:
agree, removed.

VojtechLanka commented on src/components/customTable.js:
```diff
@@ -0,0 +1,105 @@
+          console.log(this.getAttributes()['data-isCreated'])
+          console.log("this is an edit, these attributes changes: ")
+          console.log(this.changed.attributes)
+          console.log("hasHeader: "+ this.checkHeaderExists())
```
I don't think that final plugin should log something to console.

sneddobuilds replied:
agree, removed.

VojtechLanka commented on src/components/index.js:
```diff
@@ -0,0 +1,10 @@
+import CustomTable from "./customTable";
+import TableCell from "./TableCell";
```
I don't like that one file is in camel case and the other in pascal case.

VojtechLanka commented on src/onLoad.js:
```diff
@@ -0,0 +1,55 @@
+    $(function() {
+        let css =`.column-actions
+        {
+            background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none"><path stroke="%23ffffff" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M3 3h3M3 21h3m0 0h4a2 2 0 0 0 2-2V9M6 21V9m0-6h4a2 2 0 0 1 2 2v4M6 3v6M3 9h3m0 0h6m-9 6h9m3-3h3m0 0h3m-3 0v3m0-3V9"/></svg>');
```
Would it not be better to just store the images as .svg files?

sneddobuilds replied:
Maybe, tbh I'm not sure how to do that within an npm package so went with what I do know.

VojtechLanka replied:
Honestly didn't really think of that at the moment. Lets just keep it as it is.

VojtechLanka reviewed (changes requested):
Found some issues, mostly with the style of the code but there is one feature that is not working for me.

sneddobuilds commented:
I carried on with changes and have a few bits in prog, wasn't aware they would end up in the same PR tbh. Will tidy up later. Cheers.

sneddobuilds committed:
using es lint to fix code styles

sneddobuilds committed:
Rename customTable.js to CustomTable.js

sneddobuilds committed:
changing attribute names to match case

sneddobuilds committed:
Moving from attributes to traits / props

Moving from attributes to traits / props as event firing appears to work a little better, can listen for (and receive) an event for a single prop change.

Remove the modal as now using traits the settings panel holds the information. added defaults for table creation, will make these configurable.

sneddobuilds committed:
Fixing row count when header is turned on.

sneddobuilds committed:
Removing unused functions

sneddobuilds committed:
Added inline header removal

sneddobuilds committed:
Removed console.log lines

sneddobuilds committed:
Changing types due to clash with other plugins and removing class for cells

VojtechLanka commented on src/commands/index.js:
```diff
@@ -0,0 +1,427 @@
+        editor.selectRemove(selected);
+        setTimeout(function() {
+ editor.select(selected); 
+}, 50);
```
Seems like something broke the formatting. Seems to be issue with all `setTimeout`

sneddobuilds replied:
jslint did that, I can manually correct it, not familiar enough with it to get the rules to not do that.

VojtechLanka reviewed (changes requested):
Two small things.

VojtechLanka commented on src/traitsHelper.js:
```diff
@@ -0,0 +1,6 @@
+export function setValue(component, traitName, newValue) {
+    const trait = component.getTrait(traitName)
+    //const traitOptions = [{value: newValue}]
+    //trait.set('options', [...trait.get('options'), ...traitOptions])
```
Is this still needed here?

sneddobuilds replied:
traitsHelper is not actually used right now, but it will be used in the future. I can take out the file for now though.

sneddobuilds committed:
Fix formatting for the setTimeout calls.
